### PR TITLE
:sparkles: feat: replace deprecated navigation method

### LINKED ISSUE #1417: onBackPressed() has been deprecated
Our firestore quickstart makes use of the `onBackPressed()` method, which has been deprecated:

https://github.com/firebase/quickstart-android/blob/a219d94278faff75774509e5b09fb6d0f02004eb/firestore/app/src/main/java/com/google/firebase/example/fireeats/kotlin/RestaurantDetailFragment.kt#L136-L138

Since that app uses the [Navigation component](https://developer.android.com/guide/navigation/navigation-getting-started), we should probably replace that call with [`popBackStack()`](https://developer.android.com/reference/androidx/navigation/NavController#popBackStack()):

```kotlin
    private fun onBackArrowClicked() {
        findNavController().popBackStack()
    }
```

### PR DESCRIPTION
**Pull request by AlgyJr**
Replace `onBackPressed()` method to `popBackStack()`

Closes #1417

### CODE REVIEW

**Comment by thatfiredev** [comment]
@AlgyJr Thanks for working on this!
Can you please make the same change on the RestaurantDetailFragment.java ? We have the same method there.

**Comment by AlgyJr** [comment]
Sure @thatfiredev

**Comment by AlgyJr** [comment]
Done! @thatfiredev

**Comment by thatfiredev** [comment]
@AlgyJr `findNavController()` is a Kotlin extension function.

The Java equivalent should be `NavHostFragment.findNavController(this);` as shown here: https://github.com/androidx/androidx/blob/6c9916038eed0d82975c12442a6e75b907ffbfce/navigation/navigation-fragment/src/main/java/androidx/navigation/fragment/Fragment.kt#L28-L29

**Comment by AlgyJr** [comment]
And now @thatfiredev ? :)

**Comment by thatfiredev** [comment]
@AlgyJr I think you're missing the import on Java:

```java
import androidx.navigation.NavController
```

Also, we sort our imports alphabetically, so make sure you put it on the correct order at the top of the file.

**Comment by thatfiredev** [comment]
Oh, also missing the Kotlin import:

```kotlin
import androidx.navigation.fragment.findNavController
```

File: firestore/app/src/main/java/com/google/firebase/example/fireeats/java/RestaurantDetailFragment.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.fragment.app.Fragment;
+import androidx.navigation.fragment.findNavController;
```
Comment (thatfiredev) [inline firestore/app/src/main/java/com/google/firebase/example/fireeats/java/RestaurantDetailFragment.java:15]: The correct java import is `NavController`:
```suggestion
import androidx.navigation.NavController;
```

Comment (AlgyJr) [reply firestore/app/src/main/java/com/google/firebase/example/fireeats/java/RestaurantDetailFragment.java:15]: Sure, got distracted. Thanks.

File: firestore/app/src/main/java/com/google/firebase/example/fireeats/java/RestaurantDetailFragment.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.fragment.app.Fragment;
+import androidx.navigation.NavController;
```
Comment (thatfiredev) [inline firestore/app/src/main/java/com/google/firebase/example/fireeats/java/RestaurantDetailFragment.java:15]: Oops, seems like it should have been `NavHostFragment`:
```suggestion
import androidx.navigation.NavHostFragment;
```

File: firestore/app/src/main/java/com/google/firebase/example/fireeats/java/RestaurantDetailFragment.java [hunk 3]
```diff
@@ -12,6 +12,7 @@
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.fragment.app.Fragment;
+import androidx.navigation.NavHostFragment;
```
Comment (thatfiredev) [inline firestore/app/src/main/java/com/google/firebase/example/fireeats/java/RestaurantDetailFragment.java:15]:
```suggestion
import androidx.navigation.fragment.NavHostFragment;
```

**Review by thatfiredev — APPROVED** [review]
Thank you @AlgyJr !!

**Pull request merged**